tools/node-modules

```diff
@@ -56,35 +56,96 @@ EOF
     clone_from_cache "${sha}"
 }
 
-cmd_rebuild() {
-    for file in npm-shrinkwrap.json package-lock.json yarn.lock; do
-        if [ -e "${file}" ]; then
-            echo "Please delete ${file} from $(pwd) and start again"
-            exit 1
-        fi
-    done
+cmd_npm_install() {
+    if [ -f /run/.toolboxenv ]; then
+        exec flatpak-spawn --host -- "$0" npm_install "$@"
+        exit 1
+    fi
+
+    CACHE="cockpit-project-npm-cache-volume"
```

Comment:
Hmm, this feels like clutter to me. Developers don't do this very often. If you want to cache it, could we use ./tmp/npm-cache/ ?

Reply:
It certainly came in handy in the last days of testing from my not-so-fast home internet connection :)

I agree that it would be quite nice if there was a way to expire unused volumes...

Hard no on bindmounting part of the hose filesystem, though.  This is about isolation.

Reply:
Okay -- let's keep it for now then -- we can still drop the caching once this stabilized a bit.